packages/core/src/StateMachine.ts

```diff
@@ -560,13 +560,11 @@ export class StateMachine<
         return;
       }
 
-      const actorState = logic.restoreState?.(childState, _actorScope);
-
       const actorRef = createActor(logic, {
         id: actorId,
         parent: _actorScope?.self,
         syncSnapshot: actorData.syncSnapshot,
-        state: actorState,
+        state: childState,
```

Comment:
`restoreState` is already called within `createActor` ([here](https://github.com/statelyai/xstate/blob/1b6e26482344a7de2bf86e6f3a75340528500a52/packages/core/src/interpreter.ts#L211)) so we were calling it twice~ and that was creating problems